### PR TITLE
Default None Object for attributes that are meant to be None by default.

### PR DESCRIPTION
**Pull request by carycheng**

### CODE REVIEW

**Comment by boxcla** [comment]
Hi @carycheng, thanks for the pull request. Before we can merge it, we need you to sign our Contributor License Agreement. You can do so electronically here: http://opensource.box.com/cla

Once you have signed, just add a comment to this pull request saying, "CLA signed". Thanks!

File: boxsdk/object/item.py [hunk 1]
```diff
@@ -219,8 +220,11 @@ def create_shared_link(
             and has permission to set expiration dates.  Takes an RFC3339-formatted string, e.g.
             '2018-10-31T23:59:59-07:00' for 11:59:59 PM on October 31, 2018 in the America/Los_Angeles timezone.
             The time portion can be omitted, which defaults to midnight (00:00:00) on that date.
+
+            SDK_VALUE_NOT_SET defaults to an `object` to differentiate between default no unshared_at time set or the
+            intention to clear the unshared_at field.
         :type unshared_at:
-            `unicode` or None
+            SDK_VALUE_NOT_SET or `unicode`
```
Comment (Jeff-Meadows) [inline boxsdk/object/item.py]: could still be `None`

File: boxsdk/object/item.py [hunk 2]
```diff
@@ -219,6 +220,9 @@ def create_shared_link(
             and has permission to set expiration dates.  Takes an RFC3339-formatted string, e.g.
             '2018-10-31T23:59:59-07:00' for 11:59:59 PM on October 31, 2018 in the America/Los_Angeles timezone.
             The time portion can be omitted, which defaults to midnight (00:00:00) on that date.
+
+            SDK_VALUE_NOT_SET defaults to an `object` to differentiate between default no unshared_at time set or the
```
Comment (mattwiller) [inline boxsdk/object/item.py]: This probably doesn't need to be included in the public documentation for this method

File: test/unit/object/conftest.py [hunk 3]
```diff
@@ -326,7 +327,7 @@ def shared_link_password(request):
     return request.param
 
 
-@pytest.fixture(params=('2018-10-31', '2018-10-31T23:59:59-07:00', None))
+@pytest.fixture(params=('2018-10-31', '2018-10-31T23:59:59-07:00', SDK_VALUE_NOT_SET))
```
Comment (mattwiller) [inline test/unit/object/conftest.py]: Why not add `None` here?

**Pull request merged**